aws/resource_aws_datapipeline_test.go

```diff
@@ -0,0 +1,269 @@
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccAWSDataPipelineConfig(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSDataPipelineExists(resourceName, &conf),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "0"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAWSDataPipelineDisappears(conf *datapipeline.PipelineDescription) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		conn := testAccProvider.Meta().(*AWSClient).datapipelineconn
+		params := &datapipeline.DeletePipelineInput{
+			PipelineId: conf.PipelineId,
+		}
+
+		_, err := conn.DeletePipeline(params)
+		if err != nil {
+			return err
+		}
+		return waitForDataPipelineDeletion(conn, *conf.PipelineId)
+	}
+}
+
+func testAccCheckAWSDataPipelineDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).datapipelineconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_datapipeline" {
+			continue
+		}
+		// Try to find the Pipeline
+		var err error
+		resp, err := conn.DescribePipelines(
+			&datapipeline.DescribePipelinesInput{
+				PipelineIds: []*string{aws.String(rs.Primary.ID)},
+			})
+
+		if err != nil {
+			if isAWSErr(err, datapipeline.ErrCodePipelineNotFoundException, "") {
+				continue
+			} else if isAWSErr(err, datapipeline.ErrCodePipelineDeletedException, "") {
+				continue
+			}
+			return err
+		}
+		if len(resp.PipelineDescriptionList) != 0 &&
+			*resp.PipelineDescriptionList[0].PipelineId == rs.Primary.ID {
+			return fmt.Errorf("Pipeline still exists")
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckAWSDataPipelineExists(n string, v *datapipeline.PipelineDescription) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No DataPipeline ID is set")
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).datapipelineconn
+
+		opts := &datapipeline.DescribePipelinesInput{
+			PipelineIds: []*string{aws.String(rs.Primary.ID)},
+		}
+
+		resp, err := conn.DescribePipelines(opts)
+		if err != nil {
+			return err
+		}
+		if len(resp.PipelineDescriptionList) != 1 ||
+			*resp.PipelineDescriptionList[0].PipelineId != rs.Primary.ID {
+			return fmt.Errorf("DataPipeline not found")
+		}
+
+		*v = *resp.PipelineDescriptionList[0]
+		return nil
+	}
+}
+
+func testAccCheckAWSDataPipelineNotEqual(pipeline1, pipeline2 *datapipeline.PipelineDescription) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		if aws.StringValue(pipeline1.PipelineId) == aws.StringValue(pipeline2.PipelineId) {
+			return fmt.Errorf("Pipeline IDs are equal")
+		}
+
+		return nil
+	}
+}
+
+func testAccAWSDataPipelineConfig(rName string) string {
+	return fmt.Sprintf(`
+resource "aws_datapipeline" "default" {
+	name      = "%[1]s"
```

Comment:
The `%q` verb can be used to automatically quote strings:

```suggestion
  name = %[1]q
```